libpod/container_internal_linux.go

```diff
@@ -1399,3 +1411,42 @@ func (c *Container) getOCICgroupPath() (string, error) {
 		return "", errors.Wrapf(define.ErrInvalidArg, "invalid cgroup manager %s requested", c.runtime.config.CgroupManager)
 	}
 }
+
+func (c *Container) setupOverlayMounts() error {
```

Comment:
I'd prefer it if we parse the mounts when we create the container and add a field to the DB for `OverlayVolumes` and then have this work off of that instead of parsing every mount in the spec. It's messier at create time but should be faster and cleaner for `podman run`.

Reply:
Sorry, faster and cleaner for `podman start`

Reply:
Still think that leaving this in the spec and parsing it out is not a good way of doing this - mimicking named volumes (not in the spec, added to the DB and then added to the spec from there after we finish the mount) is probably better.

Reply:
We're also still rewriting c.config.Spec which is definitely not correct or safe

Reply:
c.config.Spec is still being changed - this will not work for use-cases where we have multiple commands running in the same process (e.g. APIv2 podman-system-service command)